apps/remote_control/lib/lexical/remote_control/code_intelligence/rename.ex

```diff
@@ -0,0 +1,178 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Rename do
+  alias Lexical.Ast
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Line
+  alias Lexical.Document.Position
+  alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
+  require Logger
+
+  import Line
+
+  @spec rename(Analysis.t(), Position.t(), String.t()) ::
+          {:ok, %{Lexical.uri() => [Edit.t()]}} | {:error, term()}
+  def rename(%Analysis{} = analysis, %Position{} = position, new_name) do
+    with {:ok, entity, range} <- resolve_module(analysis, position) do
```

Comment:
What do you think about replacing `resolve_module` with a new `Entity.resolve/3` API that would work like:

```suggestion
    with {:ok, entity, range} <- Entity.resolve(analysis, position, [:module, :struct]) do
```

Inside `Entity`, I think the implementation would just be:

```elixir
def resolve(%Analysis{} = analysis, %Position{} = position, kinds) when is_list(kinds) do
  case resolve(analysis, position) do
    {:error, _} = error ->
      error

    entity when is_tuple(entity) and elem(entity, 0) in kinds ->
      entity

    _ ->
      {:error, :not_found}
  end
end
```

Reply:
I think we can do this in the future, but for now, I prefer to keep it in this file, as this API is not used elsewhere.